lib/paperclip/storage/s3.rb

```diff
@@ -113,21 +113,23 @@ module Storage
 
     module S3
       def self.extended base
-        unless defined?(AWS_CLASS)
-          begin
-            require 'aws-sdk'
-            const_set('AWS_CLASS', defined?(::Aws) ? ::Aws : ::AWS)
-            const_set('AWS_BASE_ERROR',
-              defined?(::Aws) ? Aws::Errors::ServiceError : AWS::Errors::Base)
-            const_set('DEFAULT_PERMISSION',
-              defined?(::AWS) ? :public_read : :'public-read')
-
-          rescue LoadError => e
-            e.message << " (You may need to install the aws-sdk gem)"
-            raise e
-          end
-          if Gem::Version.new(AWS_CLASS::VERSION) >= Gem::Version.new(2) && Gem::Version.new(AWS_CLASS::VERSION) <= Gem::Version.new("2.0.33")
-            raise LoadError, "paperclip does not support aws-sdk versions 2.0.0 - 2.0.33.  Please upgrade aws-sdk to a newer version."
+        Thread.exclusive do
+          unless defined?(AWS_CLASS)
+            begin
+              require 'aws-sdk'
+              const_set('AWS_CLASS', defined?(::Aws) ? ::Aws : ::AWS)
+              const_set('AWS_BASE_ERROR',
+                defined?(::Aws) ? Aws::Errors::ServiceError : AWS::Errors::Base)
+              const_set('DEFAULT_PERMISSION',
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.